File: CON BDD/EUREKABANK_SOAP_DOTNET_GR06/08. SERVIDOR/Servicio/SeguridadUtil.cs
using System.Security.Cryptography;
using System.Text;

namespace SERVIDOR.Servicio
{
    public static class SeguridadUtil
    {
        public static string Sha1(string texto)
        {
            using var sha1 = SHA1.Create();
            byte[] hash = sha1.ComputeHash(Encoding.UTF8.GetBytes(texto));
            StringBuilder sb = new StringBuilder(hash.Length * 2);
            foreach (byte b in hash)
            {
                sb.Append(b.ToString("x2"));
            }
            return sb.ToString();
        }
    }
}

File: CON BDD/EUREKABANK_RESTFULL_JAVA_GR06/08. SERVIDOR/eurekabank_restful_java_gr06/src/main/java/ec/edu/monster/servicio/SeguridadUtil.java
package ec.edu.monster.servicio;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.nio.charset.StandardCharsets;

/** Utilidad de seguridad: hash SHA-1 en hexadecimal (formato de la tabla usuario). */
public final class SeguridadUtil {

    private SeguridadUtil() {
    }

    /** Devuelve el SHA-1 del texto en hexadecimal minuscula (40 caracteres). */
    public static String sha1(String texto) {
        try {
            MessageDigest md = MessageDigest.getInstance("SHA-1");
            byte[] hash = md.digest(texto.getBytes(StandardCharsets.UTF_8));
            StringBuilder sb = new StringBuilder(hash.length * 2);
            for (byte b : hash) {
                sb.append(Character.forDigit((b >> 4) & 0xF, 16));
                sb.append(Character.forDigit(b & 0xF, 16));
            }
            return sb.toString();
        } catch (NoSuchAlgorithmException e) {
            throw new IllegalStateException("SHA-1 no disponible", e);
        }
    }
}
